app/services/solar/solis_cloud_upserter.rb

```diff
@@ -0,0 +1,15 @@
+require 'dashboard'
+
+module Solar
+  class SolisCloudUpserter < BaseUpserter
+    private
+
+    def meter_attributes(details)
```

Comment:
Minor, but initially thought this was referring to MeterAttributes. Wonder if could rename, e.g. `meter_defaults`?